### ISSUE TITLE
no-shadow doesn't catch shadowing within an anonymous function

### ISSUE THREAD

**Issue by kespindler** [eslint/eslint#1436]
Eslint 0.9.2

Given a file 

```
// example1.js
var a = 5;
function who() {
    var a = 12; 
    return a + 5;
}
```

and run the command

```
eslint --no-eslintrc example1.js 
```

gives, (among others) 

```
3:8  error  a is already declared in the upper scope  no-shadow
```

While the following file

```
// example2.js
var a = 5;
var x = function () {
    var a = 12; 
    return a + 5;
}
```

has no such complaint from 

```
eslint --no-eslintrc example2.js 
```
